modules/installation-restricted-network-samples.adoc

```diff
@@ -0,0 +1,62 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_restricted_networks/installing-restricted-networks-preparations.adoc
+
+[id="installation-restricted-network-samples_{context}"]
+= Using sample imagestreams in a restricted network installation
+
+Most imagestreams in the OpenShift namespace managed by the Samples Operator
+point to images located in the Red Hat registry at link:registry.redhat.io[registry.redhat.io]. Mirroring
+will not apply to these imagestreams.
+
+The `jenkins`, `jenkins-agent-maven`, and `jenkins-agent-nodejs` imagestreams do
+come from the install payload and are managed by the Samples Operator, so no
+further mirroring procedures are needed for those imagestreams.
+
+[NOTE]
+====
+The `cli`, `installer`, `must-gather`, and `tests` imagestreams, while
+part of the install payload, are not managed by the Samples Operator. These are
+not addressed in this procedure.
+====
+
+.Prerequisites
+* Cluster administrator privileges
+
+.Procedure
+
+. Mirror images from link:registry.redhat.io[registry.redhat.io] associated with any imagestreams you need
+in the restricted network environment into one of the defined mirrors.
++
+. Add the required trusted CAs for the mirror in the cluster’s image
+configuration object:
```

Comment:
@bmcelvee since the original google doc .... in addition to the CAs. the mirrored registry quite possibly will have required auth associated with it ... we ran into the same thing with the builds in disconnected environment support.

With that, I believe the user will need to add a pull secret to the openshift namespace with the mirrored registries credentials.  That pull secret should be automatically leveraged when the OCM initiates image imports for the image streams they have mirrored.

@bparees @mtrmac FYI ^^ in case I've misinterpreted something.

Reply:
Will this be sufficient to cover the pull secret requirement?

https://bz1738476--ocpdocs.netlify.com/openshift-enterprise/latest/installing/installing_restricted_networks/installing-restricted-networks-preparations.html#installation-local-registry-pull-secret_installing-restricted-networks-preparations

It's a module included in the assembly that this module is going into.

Reply:
I would expect the user to provide a pullsecret to the install, that can pull from their mirror (because all the infra pods are going to have to pull their image from there).

So the pull-secret the samples operator sees and copies to `openshift` should already have the necessary creds for this.

Reply:
(but certainly if for some reason they mirror the sample content to a different registry than what they mirrored the OCP images themselves to, they would need to setup the additional secret in the `openshift` namespace)

Reply:
If they put the registry.redhat.io content in the same mirror as the install payload, I would agree @bparees   

Is there not the notion of creating more than one mirror registry (I'm asking more than speaking from knowledge)?  ... separate from the mirror used for the install payload ... I see `Copy image to multiple locations` from `oc help image mirror`.

Reply:
> Is there not the notion of creating more than one mirror registry

yeah, you've anticipated my second comment above.

Reply:
@gabemontero @bmcelvee this ties back to my "pull secrets docs" email.  If we had good documentation on how to setup pullsecrets for use by the samples operator, you could just point there and say "make sure the samples operator has creds to import from this mirror registry (which it may already thanks to the pull-secret you used at install-time)"

Reply:
https://jira.coreos.com/browse/OSDOCS-715

Reply:
I checked in with Gabe, and the "Creating a pull secret for your mirror registry" procedure in the same assembly that this procedure is being included in is sufficient (at least for now). I also updated the prerequisites for this procedure accordingly. 

For OSDOCS-715, I bumped it up in priority so I can get to it sooner.